core/src/main/scala/io/delta/tables/execution/DeltaConvert.scala

```diff
@@ -29,12 +30,14 @@ trait DeltaConvertBase {
       tableIdentifier: TableIdentifier,
       partitionSchema: Option[StructType],
       deltaPath: Option[String]): DeltaTable = {
-    val cvt = ConvertToDeltaCommand(tableIdentifier, partitionSchema, deltaPath)
-    cvt.run(spark)
-    if (cvt.isCatalogTable(spark.sessionState.analyzer, tableIdentifier)) {
-      DeltaTable.forName(spark, tableIdentifier.toString)
-    } else {
-      DeltaTable.forPath(spark, tableIdentifier.table)
+    DeltaTableUtils.withActiveSession(spark) {
+      val cvt = ConvertToDeltaCommand(tableIdentifier, partitionSchema, deltaPath)
```

Comment:
It would be nice to have tests for other operations than MERGE but it's probably going to take a bit of effort to figure out cases that break today for each operation. I don't think we should block that fix on that though, it's pretty clear by now that the session needs to be pinned before running commands